pkg/client/openstack/kluster/client.go

```diff
@@ -190,94 +202,249 @@ func (c *klusterClient) SetSecurityGroup(sgName, nodeID string) (err error) {
 	return secgroups.AddServer(c.ComputeClient, nodeID, sgName).ExtractErr()
 }
 
-func (c *klusterClient) EnsureKubernikusRuleInSecurityGroup(kluster *v1.Kluster) (created bool, err error) {
-	sgName := kluster.Spec.Openstack.SecurityGroupName
-	page, err := securitygroups.List(c.NetworkClient, securitygroups.ListOpts{Name: sgName}).AllPages()
+type portListOpts struct {
+	ProjectID      string `q:"project_id"`
+	SecurityGroups string `q:"security_groups"`
+}
+
+func (opts portListOpts) ToPortListQuery() (string, error) {
+	q, err := gophercloud.BuildQueryString(opts)
+	params := q.Query()
+	q = &url.URL{RawQuery: params.Encode()}
+	return q.String(), err
+}
+
+func getLoadbalancerSecurityGroup(kluster *v1.Kluster, client *gophercloud.ServiceClient) (*securitygroups.SecGroup, error) {
```

Comment:
As we are getting two different security groups in this controller maybe its worthwhile to have a single `getSecurityGroupByName(name string, client *gophercloud.ServiceClient)` function instead if this special purpose function?

Reply:
Yeah, makes sense.